build.sbt

```diff
@@ -18,32 +18,23 @@ ThisBuild / scalafixAll / skip := tlIsScala3.value
 ThisBuild / ScalafixConfig / skip := tlIsScala3.value
 ThisBuild / Test / scalafixConfig := Some(file(".scalafix.test.conf"))
 
-ThisBuild / githubWorkflowOSes := Seq("ubuntu-22.04")
-
-ThisBuild / githubWorkflowBuildPreamble +=
-  WorkflowStep.Run(
-    List("/home/linuxbrew/.linuxbrew/bin/brew install s2n"),
-    name = Some("Install s2n"),
-    cond = Some("startsWith(matrix.project, 'rootNative')"),
+ThisBuild / githubWorkflowJobSetup ++= Seq(
+  WorkflowStep.Use(
+    UseRef.Public("cachix", "install-nix-action", "v17"),
+    name = Some("Install Nix"),
+    params = Map("extra_nix_config" -> "access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}"),
   )
-val isLinux = {
-  val osName = Option(System.getProperty("os.name"))
-  osName.exists(_.toLowerCase().contains("linux"))
-}
-val isMacOs = {
-  val osName = Option(System.getProperty("os.name"))
-  osName.exists(_.toLowerCase().contains("mac"))
-}
+)
+
+ThisBuild / githubWorkflowSbtCommand := "nix develop -c sbt"
```

Comment:
The sbt derivation comes with its own jre attribute to provide its default JDK, which is why it's always running with 8.  Instead of overstuffing the dev shell with multiple Java versions and faffing about with environment variables, I just created three dev shells in #6728.  The default is still the lowest supported JDK, but three options are provided for CI.

Reply:
Super thanks for that!!